### DISCUSSION TITLE
TSC Meeting Minutes October 2021.

### DISCUSSION THREAD

**Discussion by lgirdwood** [thesofproject/sof#7397] [Team Posts]
**Attendees.**
@lgirdwood @plbossart @dbaluta @mwasko @beatabaranowska @cujomalainey @bzhg @harshapriya-n @balakishorepati @bhiregou @sunilkumardommati @MaureenHelm @yaochunhung @kuanhsuncheng

**Items**
I think I've captured most of the discussion, but as usual I may have made mistakes or omitted some details. TSC members please reply with any corrections and I can add here if agreed for the record. Thanks !

1) **v1.9 release.** Teams have different development models. Intel, AMD, & Mediatek tend to work on FW/Linux HEAD, whilst NXP tend to work on an older main/stable branch and kernel and then forward port. AR for @lgirdwood and @plbossart to look at development model whether we have a commit /revert model or a -next style branch. AMD, Mediatek and NXP planning CI integration, but date is TBD.

2) **Next release is v2.0**, but with a smaller development window due to Thanksgiving and Christmas holidays. RC1 date is mid November for v2.0 mid December.

3) **Windows** support is making progress. Windows can parse the manifest and boot firmware now. Next steps are fundamental playback and capture on Windows which may make the v2.0 release if progress is good. Windows CI to be added in CI pool soon, date TBD.

4) **Zephyr** also making good progress. Intel has ported 4 platforms and NXP 3 platforms. Agree to deprecate xtos over time and make Zephyr primary RTOS. Abstraction also needed to support FreeRTOS which should be easier for common RTOS functions where Zephyr and FreeRTOS export APIs (like scheduling) but more complex where FreeRTOS has no API (like audio and most drivers). Intel to make Zephyr primary RTOS for TGL/ADL in V2.0 Linux release.

5) **Multicore** complicated on some xtensa devices due to manual cache management (i.e. its done by FW) and this can be difficult for developers to maintain coherence. Intel updating some core APIs to allow shared object access with locking over multiple cores. AMD looking at changes required for heterogeneous multicore (potentially with different ISAs) and how tooling can be used to generate 2 binaries stitched into 1 image (Zephyr looked at this in the past and will probably proceed with looking at this in more detail).

6) **Codec adapter**. Some complications still exist around pipelines with mixed PCM and compressed around passing parameters and selecting codec (in multicodec components). Intermediate PR pending to fix for v2.0 but more work needs done to have a scalable fix (i.e. work around mixer, mux, demux to enable them as static endpoints for connected pipelines). Codec discovery is also limited today alongside no gapless compressed playback. AR for @plbossart and @dbaluta to further investigate and propose solution.

7) **clang compiler** transition code to support clang builds. Agreed with everyone, no objections.

8) **Topology.** Today FW and topology released together in sof-bin, but this does not allow topology features or configuration to align with kernel features (for any particular kernel version). Topology probably needs a separate repo or align with ALSA repo that may help AND/OR a qualifier mechanism for capabilities sync and negotiation with kernel. AR @plbossart to describe a suitable kernel qualifier mechanism for further discussion on how this can be added into topology.

**Comment by kuanhsuncheng** [comment]
HI Liam,

About item1, because sof of mtk platform is in chromebook product, we tend to work on FW/Linux HEAD following Google policy.

And we also are planning CI integration. but it still needs some time to investigate, the date is TBD

thanks.

**Comment by lgirdwood** [comment]
Thanks @kuanhsuncheng now updated.

**Comment by yaochunhung** [comment]
@lgirdwood I have one question about qemu. Now we are developing qemu, could you please tell me which branch I should base on to submit? Currently I use sof-v4.2. Thanks

**Comment by marc-hb** [comment]
There is a more recent sof-v5.2.0 branch. It's not used in CI yet because it's not stable enough yet but you should try it first.

**Comment by yaochunhung** [comment]
@marc-hb  Got it. We will use v5.2. Thanks.
